sim/paladin/item_sets_pve.go

```diff
@@ -443,78 +204,482 @@ var ItemSetWilfullJudgement = core.NewItemSet(core.ItemSet{
+			//Damage multiplier is Additive with Infusion of Light rather than multiplicitive
+			spell.DamageMultiplierAdditive += 0.5
+		}
+	})
+}
+
+func (paladin *Paladin) applyPaladinZG5P() {
+	if duplicateBonusCheck(paladin, PaladinZG5P, "PaladinZG5P") {
+		return
+	}
+
+	paladin.RegisterAura(core.Aura{
+		Label: "S03 - Item - ZG - Paladin - Caster 5P Bonus",
+		OnInit: func(aura *core.Aura, sim *core.Simulation) {
+			for _, spell := range paladin.exorcism {
+				spell.CD.Duration -= time.Second * 3
+				spell.DamageMultiplierAdditive += 0.5
+			}
+		},
+	})
+}
+
+func (paladin *Paladin) applyPaladinRAQ3P() {
+	if duplicateBonusCheck(paladin, PaladinRAQ3P, "PaladinRAQ3P") {
+		return
+	}
+
+	aura := core.Aura{
+		Label: "S03 - Item - RAQ - Paladin - Retribution 3P Bonus",
+		OnInit: func(aura *core.Aura, sim *core.Simulation) {
+
 			if !paladin.hasRune(proto.PaladinRune_RuneHandsCrusaderStrike) {
 				return
 			}
 
-			paladin.RegisterAura(core.Aura{
-				Label: "S03 - Item - RAQ - Paladin - Retribution 3P Bonus",
-				OnInit: func(aura *core.Aura, sim *core.Simulation) {
-					originalApplyEffects := paladin.crusaderStrike.ApplyEffects
-					extraApplyEffects := paladin.judgement.ApplyEffects
-
-					// Wrap the apply Crusader Strike ApplyEffects with more Effects
-					paladin.crusaderStrike.ApplyEffects = func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
-						originalApplyEffects(sim, target, spell)
-						consumeSealsOnJudgeSaved := paladin.consumeSealsOnJudge // Save current value
-						paladin.consumeSealsOnJudge = false                     // Set to not consume seals
-						if paladin.currentSeal.IsActive() {
-							extraApplyEffects(sim, target, paladin.judgement)
-						}
-						paladin.consumeSealsOnJudge = consumeSealsOnJudgeSaved // Restore saved value
-					}
-				},
-			})
+			originalApplyEffects := paladin.crusaderStrike.ApplyEffects
+			extraApplyEffects := paladin.judgement.ApplyEffects
+
+			// Wrap the apply Crusader Strike ApplyEffects with more Effects
+			paladin.crusaderStrike.ApplyEffects = func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+				originalApplyEffects(sim, target, spell)
+				consumeSealsOnJudgeSaved := paladin.consumeSealsOnJudge // Save current value
+				paladin.consumeSealsOnJudge = false                     // Set to not consume seals
+				if paladin.currentSeal.IsActive() {
+					extraApplyEffects(sim, target, paladin.judgement)
+				}
+				paladin.consumeSealsOnJudge = consumeSealsOnJudgeSaved // Restore saved value
+			}
 		},
-	},
-})
+	}
+
+	paladin.RegisterAura(aura)
+}
+
+func duplicateBonusCheck(agent core.Agent, bonusID int32, bonusString string) bool {
+	paladin := agent.(PaladinAgent).GetPaladin()
+
+	if paladin.GetAuraByID(core.ActionID{SpellID: bonusID}) != nil {
+		return true // Do not apply bonus aura more than once (Due to Soul)
+	}
+
+	paladin.RegisterAura(core.Aura{
+		Label:    bonusString,
+		ActionID: core.ActionID{SpellID: bonusID},
+	})
```

Comment:
I'm not a big fan of how this creates a separate aura from other set bonus auras created in the individual bonus functions. It ends up creating duplicates vs the way that I just used labels in other classes' code
https://github.com/wowsims/sod/blob/4b20b13717962d5075f47df57f64df8ddf3cbb4d/sim/shaman/item_sets_pve_phase_4.go#L164-L175